links/gql_websocket_link/lib/src/graphql_transport_ws.dart

```diff
@@ -813,6 +815,13 @@ class _ConnectionState {
             }
             // parseMessage(msg!, reviver: options.jsonMessageReviver);
             if (!isOpen) return;
+
+            // wait for next or error message (result) to be processed before process complete message
+            while (message is CompleteMessage &&
+                nextOrErrorMsgWaitMap.containsKey(message.id)) {
+              await Future.delayed(const Duration(milliseconds: 100));
+            }
```

Comment:
could this potential endless loop be replaces using a completed? should there be a timeout to avoid hanging forever?